src/api.rs

```diff
@@ -293,7 +293,7 @@ fn validate_ref (ref_name: &String, req: &HttpRequest) -> Result<(),ApiError>
             }
             req.has_token_prefix(ref_parts[1])
         },
-        _  => Err(ApiError::BadRequest(format!("Invalid ref_name {}", ref_name))),
+        _  => Ok(()), // Err(ApiError::BadRequest(format!("Invalid ref_name {}", ref_name))),
```

Comment:
I think rather than completely removing the check, it would be better to add a configuration option to disable it.

Reply:
Or maybe make it non-fatal, but still logged?

Reply:
I don't think allowing any refs at all is a good idea. Clients can then e.g. replace appstream/x86_64 or other internal "trusted" refs. Allowing some prefix seems better.